packages/suite/src/views/wallet/send/Outputs/TokenSelect/TokenAddress.tsx

```diff
@@ -0,0 +1,132 @@
+import { HiddenPlaceholder } from 'src/components/suite/HiddenPlaceholder';
+import { Icon, Link, Text, TextProps } from '@trezor/components';
+import { useMemo, useState } from 'react';
+import styled, { css, useTheme } from 'styled-components';
+import { borders, spacingsPx } from '@trezor/theme';
+
+const IconWrapper = styled.div`
+    display: none;
+    padding: ${spacingsPx.xxxs};
+    border-radius: ${borders.radii.xxxs};
+    margin-left: ${spacingsPx.xxs};
+    background-color: ${({ theme }) => theme.iconSubdued};
+    height: 16px;
+    align-items: center;
+    justify-content: center;
+
+    &:hover {
+        opacity: 0.7;
+    }
+`;
+
+const onHoverTextOverflowContainerHover = css`
+    border-radius: ${borders.radii.xxxs};
+    background-color: ${({ theme }) => theme.backgroundSurfaceElevation2};
+    outline: ${borders.widths.large} solid ${({ theme }) => theme.backgroundSurfaceElevation2};
+    z-index: 3;
+
+    ${IconWrapper} {
+        display: flex;
+    }
+`;
+
+const TextOverflowContainer = styled.div<{ $shouldAllowCopy?: boolean }>`
+    position: relative;
+    display: inline-flex;
+    align-items: center;
+    max-width: 100%;
+    overflow: hidden;
+    cursor: ${({ $shouldAllowCopy }) => ($shouldAllowCopy ? 'pointer' : 'cursor')};
+    user-select: none;
+
+    ${({ $shouldAllowCopy }) =>
+        $shouldAllowCopy &&
+        css`
+            @media (hover: none) {
+                ${onHoverTextOverflowContainerHover}
+            }
+
+            &:hover,
+            &:focus {
+                ${onHoverTextOverflowContainerHover}
+            }
+        `}
+`;
+
+interface TokenAddressProps {
+    tokenExplorerUrl?: string;
+    tokenContractAddress: string;
+    shouldAllowCopy?: boolean;
+    typographyStyle?: TextProps['typographyStyle'];
+    variant?: TextProps['variant'];
+    onCopy: () => void;
+}
+
+// This is needed because icon interferes with pointer events of Select
+// eslint-disable-next-line local-rules/no-override-ds-component
+const IconWithNoPointer = styled(Icon)`
+    pointer-events: none;
+`;
+
+export const TokenAddress = ({
```

Comment:
Are we okay that we have 2 components doing almost the same? `IOAddress` and this one `TokenAddress`.